src/components/NFTCollection/NFTCard.tsx

```diff
@@ -0,0 +1,202 @@
+                            src={nft.image}
+                            onError={() => setIsError(true)}
+                        ></video>
+                    ) : (
+                        <ImageStyled
+                            src={nft.image}
+                            onError={() => setIsError(true)}
+                        />
+                    )
+                ) : (
+                    <Illustration logo="lazyNft" width="100%" height="200px" />
+                )}
+                <DivStyledCardContent>
+                    <a
+                        href={`https://blockscan.com/address/${tokenAddress}`}
+                        target="_blank"
+                        referrerPolicy="no-referrer"
+                    >
+                        <Typography variant="caption14">
+                            <span>{getEllipsisTxt(tokenAddress, 4)}</span>
+                        </Typography>
+                    </a>
+                    <Typography variant="caption14">ETH</Typography>
+                </DivStyledCardContent>
+                <DivStyledCardContent>
+                    <Typography variant="body16">
+                        {nft.name && nft.name !== '' ? nft.name : '#' + tokenId}
+                    </Typography>
+                    <Typography variant="body16">NA</Typography>
+                </DivStyledCardContent>
+                <DivStyledCardFooter>
+                    <div style={{ marginLeft: 'auto' }}>
+                        <PopoverDropdown
+                            position="bottom"
+                            parent={
+                                <Icon
+                                    key="4"
+                                    svg={iconTypes.info}
+                                    fill={color.grey}
+                                    size={20}
+                                />
+                            }
+                            children={[
+                                <a
+                                    href={`http://opensea.io/assets/${tokenAddress}/${tokenId}`}
+                                    target="_blank"
+                                    referrerPolicy="no-referrer"
+                                >
+                                    <PopoverElement
+                                        key="0"
+                                        height={30}
+                                        width={150}
+                                        text={'Buy on OpenSea'}
+                                        icon={iconTypes.link}
+                                        textSize={10}
+                                        backgroundColor={'transparent'}
+                                        textColor={color.white}
+                                    />
+                                </a>,
+                                <a
+                                    href={`https://blockscan.com/address/${tokenAddress}`}
```

Comment:
you could dynamically fetch the correct block explorer by the chainId (hex)
check this
https://github.com/web3ui/web3uikit/blob/master/src/web3utils/networks.ts

Reply:
Thanks for this 🤩